src/lib/containers/works_for_you.tsx

```diff
@@ -22,31 +22,46 @@ import Notification from "../components/works_for_you/notification"
 
 import colors from "../../data/colors"
 
-interface Props extends RelayProps {}
+const PageSize = 10
+const PageEndThreshold = 1000
+
+interface Props extends RelayProps {
+  relay: any;
+}
 
 interface State {
   dataSource: ListViewDataSource | null
   sideMargin: number
   topMargin: number
+  fetchingNextPage: boolean
+  completed: boolean
 }
 
 export class WorksForYou extends React.Component<Props, State> {
   constructor(props) {
     super(props)
 
     const edges = props.me.notifications_connection.edges
-    const dataSource = edges.length && new ListView.DataSource({
-                                      rowHasChanged: (row1, row2) => row1 !== row2,
-                                    }).cloneWithRows(edges.map((edge) => edge.node))
+    const dataSource = edges.length && new ListView.DataSource({ rowHasChanged: (row1, row2) => row1 !== row2 })
```

Comment:
Just saw this! Yeah - the `FlatList` is kind of shaky and doesn't do well with pagination just yet. I think with a bit more time it'll have better documentation and/or I can take a look at its underlying code and see what's up with it. But it was more trouble than it's worth for now 🤷🏻‍♀️